File: python/examples/hello.py
# Minimal Hello World example
import asyncio
import os
from dotenv import load_dotenv
from claude_agent_sdk import query

load_dotenv()
CLAUDE_PATH = os.getenv('CLAUDE_BRIDGE_PATH', './claude_bridge.py')

# Simple query example using the functional API
async def main():
    try:
        query_stream = await query({
            "prompt": "Hello! What is your name?",
            "options": {
                "path_to_claude_code_executable": CLAUDE_PATH,
                "agent": "friendly-assistant",
                "agents": {
                    "friendly-assistant": {
                        "description": "A friendly assistant",
                        "prompt": "You are a friendly assistant.",
                    }
                }
            }
        })

        async for message in query_stream:
            print(f"Message: {message}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    asyncio.run(main())

File: typescript/examples/hello.ts
// Minimal Hello World example
// Minimal Hello World example
import { query } from '@anthropic-ai/claude-agent-sdk';
import * as dotenv from 'dotenv';

dotenv.config();
const CLAUDE_PATH = process.env.CLAUDE_BRIDGE_PATH || './claude_bridge.py';

// Simple query example using the functional API
async function main() {
  try {
    const queryStream = await query({
      prompt: 'Hello! What is your name?',
      options: {
        pathToClaudeCodeExecutable: CLAUDE_PATH,
        agent: 'friendly-assistant',
        agents: {
          'friendly-assistant': {
            description: 'A friendly assistant',
            prompt: 'You are a friendly assistant.',
          }
        }
      }
    });

    for await (const message of queryStream) {
      console.log('Message:', message);
    }
  } catch (error) {
    console.error('Error:', error);
  }
}

main();
